### ISSUE TITLE
Some php packages has been pointed to 8.3, but others not

### ISSUE THREAD

**Issue by AnnAngela** [oerdnj/deb.sury.org#2062]
**Frequently asked questions**

- [x] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)
- [x] I have looked at the list of the existing issues (including closed issues) and searched if my issue has been already reported
- [x] I have tried to resolve the issue myself and will describe what I did in clear and consise manner

**Describe the bug**

Some php packages has been pointed to php 8.3, but others pointed to php 8.2. 

The inconsistency can lead to many problems, and it is hoped that a resolution can be achieved.

**To Reproduce**

See: https://github.com/AnnAngela/AnnAngela/actions/runs/7176247067/job/19540829695

**Your understanding of what is happening**

Maybe the lack of building? <https://launchpad.net/~ondrej/+archive/ubuntu/php/+builds?build_text=&build_state=all>

**Package(s) (please complete the following information):**

Packages pointed to php 8.2:
- `php-bcmath`
- `php-bz2`
- `php-cgi`
- `php-cli`
- `php-curl`
- `php-dev`
- `php-enchant`
- `php-excimer`
- `php-fpm`
- `php-gd`
- `php-gmp`
- `php-imap`
- `php-interbase`
- `php-intl`
- `php-ldap`
- `php-lz4`
- `php-mbstring`
- `php-msgpack`
- `php-mysql`
- `php-odbc`
- `php-pgsql`
- `php-phpdbg`
- `php-pinba`
- `php-propro`
- `php-pspell`
- `php-readline`
- `php-snmp`
- `php-soap`
- `php-sqlite3`
- `php-sybase`
- `php-tidy`
- `php-xml`
- `php-zip`

Packages pointed to php 8.3:
- `php-amqp`
- `php-apcu`
- `php-ast`
- `php-decimal`
- `php-ds`
- `php-gearman`
- `php-gmagick`
- `php-gnupg`
- `php-grpc`
- `php-http`
- `php-igbinary`
- `php-imagick`
- `php-inotify`
- `php-mailparse`
- `php-maxminddb`
- `php-memcache`
- `php-memcached`
- `php-mongodb`
- `php-oauth`
- `php-pcov`
- `php-protobuf`
- `php-ps`
- `php-psr`
- `php-raphf`
- `php-rdkafka`
- `php-redis`
- `php-rrd`
- `php-smbclient`
- `php-solr`
- `php-ssh2`
- `php-stomp`
- `php-swoole`
- `php-uopz`
- `php-uploadprogress`
- `php-uuid`
- `php-vips`
- `php-xdebug`
- `php-xhprof`
- `php-xmlrpc`
- `php-yac`
- `php-yaml`
- `php-zmq`
- `php-zstd`

**Comment by AnnAngela** [comment]
@oerdnj hello, sorry for pinging, but I saw there were some successfully builds ([a build for php8.3 on jammy amd64](https://launchpad.net/~ondrej/+archive/ubuntu/php/+build/27582438), e.g.), but the packages listed above [are still targeted to php8.2](https://github.com/AnnAngela/AnnAngela/actions/runs/7347605507/job/20004314249#step:4:46), anything wrong?

**Comment by vnikolay** [comment]
Hello, I can't install packages intl, gd. In cli it says can't find them. Others common pacages installed well. How can I install them?

**Comment by danboid** [comment]
I cannot install php-pgsql under Ubuntu 22.04 when I have the sury repos enabled.

I would like to use the newer apache from the sury repos but use the php from the official Ububtu repos (which lets me install php-pgsql). Is that possible?

I will have to remove the sury repos from our web servers until this is fixed as working php postgres support is a must.

**Comment by oerdnj** [comment]
The versions will be updated when the packages itself get a new update. It would be kind of waste to update the packages to to update the default. You should use the `phpX.Y-<version>` for a specific PHP version anyway.

**Comment by MatthiasKuehneEllerhold** [comment]
I dont know if this is related or not:

`php8.2-ldap` "provides" `php-ldap`, so I dont need to install `php-ldap` directly but can just install the 8.2 version.
The same is not true for `php-mysql`. If package depends on `php-mysql` I have to install this dep-only package and therefor the 8.3 version, even if I want to use the 8.2 version. So I have both mysql versions installed.

TL;DR: Could you please add "Provides: php-mysql" to `php8.2-mysql` too like you did with `php8.2-ldap`? Thanks!

**Comment by oerdnj** [comment]
> TL;DR: Could you please add "Provides: php-mysql" to php8.2-mysql too like you did with php8.2-ldap? Thanks!

No, see https://github.com/oerdnj/deb.sury.org/issues/1836 for explanation.

**Comment by MatthiasKuehneEllerhold** [comment]
So if `mysql` was removed in 7.4, whats in "php8.3-mysql"? 
There is no "mysqli" or "pdo" package (at least not for debian). Which would be the correct depends then? Nothing because PDO and mysqli is already in php8.2-common ?
